server/src/internalClusterTest/java/org/elasticsearch/indexing/IndexActionIT.java

```diff
@@ -52,51 +49,9 @@ public void testAutoGenerateIdNoDuplicates() throws Exception {
             }
             indexRandom(true, builders);
             logger.info("verifying indexed content");
-            int numOfChecks = randomIntBetween(8, 12);
+            int numOfChecks = randomIntBetween(16, 24);
             for (int j = 0; j < numOfChecks; j++) {
-                try {
-                    logger.debug("running search with all types");
-                    assertResponse(prepareSearch("test"), response -> {
-                        if (response.getHits().getTotalHits().value() != numOfDocs) {
-                            final String message = "Count is "
-                                + response.getHits().getTotalHits().value()
-                                + " but "
-                                + numOfDocs
-                                + " was expected. "
-                                + ElasticsearchAssertions.formatShardStatus(response);
-                            logger.error("{}. search response: \n{}", message, response);
-                            fail(message);
-                        }
-                    });
-                } catch (Exception e) {
-                    logger.error("search for all docs types failed", e);
-                    if (firstError == null) {
-                        firstError = e;
-                    }
-                }
-                try {
-                    logger.debug("running search with a specific type");
-                    assertResponse(prepareSearch("test"), response -> {
-                        if (response.getHits().getTotalHits().value() != numOfDocs) {
-                            final String message = "Count is "
-                                + response.getHits().getTotalHits().value()
-                                + " but "
-                                + numOfDocs
-                                + " was expected. "
-                                + ElasticsearchAssertions.formatShardStatus(response);
-                            logger.error("{}. search response: \n{}", message, response);
-                            fail(message);
-                        }
-                    });
-                } catch (Exception e) {
-                    logger.error("search for all docs of a specific type failed", e);
-                    if (firstError == null) {
-                        firstError = e;
-                    }
-                }
-            }
-            if (firstError != null) {
-                fail(firstError.getMessage());
+                assertHitCount(prepareSearch("test"), numOfDocs);
```

Comment:
what the rationale behind the changes in this test? I get confused because I thought the code fix below would be enough to fix this test failure that gets unmuted in the same PR.

Reply:
Ah sorry I had this cleanup in my reproducer branch and kept it because the test was super dirty from before with the types removal leftover etc. Let me revert this real quick, I'll open a separate test cleanup :)